conf/route_control.py

```diff
@@ -3,528 +3,792 @@
+        self, netlink_message: dict, action: str
+    ) -> None:
+        """Listens for netlink events and handles them.
+
+        Args:
+            netlink_message (dict): The netlink message.
+            action (str): The action.
+        """
+        route_entry = self.route_parser.parse(netlink_message)
+        if action == NEW_ROUTE_ACTION and route_entry:
+            logger.info("%s event received.", NEW_ROUTE_ACTION)
+            with self.lock:
+                self._handle_new_route_entry(route_entry)
+
+        elif action == DELETE_ROUTE_ACTION and route_entry:
+            logger.info("%s event received.", DELETE_ROUTE_ACTION)
+            with self.lock:
+                self._del_route(route_entry)
+
+        elif action == NEW_NEIGHBORS_ACTION:
+            logger.info("%s event received.", NEW_NEIGHBORS_ACTION)
+            with self.lock:
+                self._parse_new_neighbor(netlink_message)
+
+    def _ping_missing_entries(self):
+        """Pings missing entries every 10 seconds.
+        The goal is to populate the ARP cache.
+        If the target host does not respond it will be pinged again.
+        """
+        while True:
+            with self.lock:
+                missing_arp_entries = list(self.unresolved_arp_queries_cache.keys())  # noqa: E501
+                logger.info("Missing ARP entries: %s", missing_arp_entries)
+            for ip in missing_arp_entries:
+                try:
+                    logger.info("Pinging %s", ip)
+                    send_ping(ip)
+                except Exception as e:
+                    logger.exception("Error when pinging %s: %s", ip, e)
+            logger.info("Finished pinging missing ARP entries. Sleeping...")
+            time.sleep(10)
+
+    def _probe_addr(self, route_entry: RouteEntry) -> None:
+        """Probes the MAC address of a neighbor.
+        Pings the neighbor to trigger the update of the ARP table.
+
+        Args:
+            neighbor (NeighborEntry): The neighbor entry.
+        """
+
+        self.unresolved_arp_queries_cache[
+            route_entry.next_hop_ip
+        ] = route_entry
+        logger.info("Adding entry %s in arp probe table", route_entry)
+        if not validate_ipv4(route_entry.next_hop_ip):
+            logger.error(
+                "The IP address %s is invalid", route_entry.next_hop_ip
+            )
+            return
+        logger.info("%s is a valid IPv4 address", route_entry.next_hop_ip)
+        send_ping(route_entry.next_hop_ip)
+
+    def _parse_new_neighbor(self, netlink_message: dict) -> None:
+        """Handle new neighbor event.
+
+        Args:
+            netlink_message (dict): The netlink message.
+        """
+        attr_dict = dict(netlink_message["attrs"])
+        unresolved_next_hop = self.unresolved_arp_queries_cache.get(
+            attr_dict[DESTINATION_IP]
+        )
+        gateway_mac = attr_dict[LINK_LAYER_ADDRESS]
+        if unresolved_next_hop:
```

Comment:
```suggestion
        route_entry = self.unresolved_arp_queries_cache.get(
            attr_dict[DESTINATION_IP]
        )
        gateway_mac = attr_dict[LINK_LAYER_ADDRESS]
        if route_entry:
```

If we find the entry in the cache, it is not an unresolved next hop, since we just received the MAC address.

Reply:
Done.